config/configtls/configtls.go

```diff
@@ -0,0 +1,110 @@
+// Copyright 2020, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package configtls
+
+import (
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"io/ioutil"
+	"path/filepath"
+)
+
+// TLSConfig exposes the common client and server TLS configurations.
+// Note: Since there isn't anything specific to a server connection. Components
+// with server connections should use TLSConfig.
+type TLSConfig struct {
+	// Path to the CA cert. For a client this verifies the server certificate.
+	// For a server this verifies client certificates. If empty uses system root CA.
+	// (optional)
+	CAFile string `mapstructure:"ca_file"`
+	// Path to the TLS cert to use for TLS required connections. (optional)
+	CertFile string `mapstructure:"cert_file"`
+	// Path to the TLS key to use for TLS required connections. (optional)
+	KeyFile string `mapstructure:"key_file"`
+}
+
+// TLSClientConfig contains TLS configurations that are specific to client
+// connections in addition to the common configurations. This should be used by
+// components configuring TLS client connections.
+type TLSClientConfig struct {
```

Comment:
I would use `TLSClientSettings`. It seems that we use as suffix `Config` for interfaces and `Settings` for concrete structs (or at least that is my impression, I may be wrong). But also the name is consistent with grpc :)